src/main/java/org/apache/maven/plugin/gpg/AbstractGpgSigner.java

```diff
@@ -193,9 +195,9 @@ public File generateSignatureForArtifact( File file )
             signature = new File( signatureDirectory, file.getName() + SIGNATURE_EXTENSION );
         }
 
-        if ( signature.exists() )
+        if ( signature.exists() && signature.delete() ) 
```

Comment:
This seems unrelated to the description.
Can you create a soste separate patch?

Reply:
I also have added some changes to reduce complexity, as suggested by SonarQube. Some methods have very high complexity, those are still under development. What is "a soste separate patch" ?

Reply:
Sorry, 'soste' was a typo, due to auto completion of my browser.

We should keep a patch focused on only just one issue: reformat code, fix a bug....

Overall your patch is more a code clean up. But in this line you are changing the behavior, so I am asking you to file a separate issue